ir/shared/src/main/scala/org/scalajs/ir/Serializers.scala

```diff
@@ -41,6 +41,14 @@ object Serializers {
    */
   final val IRMagicNumber = 0xCAFE4A53
 
+  /** A regex for a compatible stable binary IR version from which we may need
+   *  to migrate things with hacks.
+   */
+  private val CompatibleStableIRVersionRegex = {
+    val prefix = java.util.regex.Pattern.quote(ScalaJSVersions.binaryCross + ".")
+    new scala.util.matching.Regex(prefix + "(\\d+)")
+  }
```

Comment:
I'm a bit uneasy with this regex that is only tested via our binary compat linker test.

Do you think we could make `Hacks` `private[ir]` and write some basic unit tests for `useBelow`?

Reply:
Good point. I added unit tests for that.